front/src/applications/operationalStudies/components/Project/PictureUploader.tsx

```diff
@@ -181,16 +184,31 @@ function PicturePlaceholderButtons({ setTempProjectImage, safeWord }: PropsButto
 
 export default function PictureUploader({ image, setTempProjectImage, tempProjectImage }: Props) {
   const [isValid, setIsValid] = useState<boolean>(true);
+  const { t } = useTranslation('operationalStudies/project');
   const safeWord = useSelector(getUserSafeWord);
-  const handleUpload = async (file?: File) => {
-    if (file && file.type.startsWith('image/')) {
+  const dispatch = useDispatch();
+
+  const handleUpload = useCallback(async (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
Don't really have the experience for that but is this function so heavy that we need a useCallback here ? (as there is nothing in the dependancy array)

Reply:
`useCallback` is not about heavy computation, but about optimizing the re-rendering of components. When you do `onChange={() => {...}}`, it creates a new function and that changes the references of the props, consequently causing the component to re-render. useCallback prevents this by keeping the reference of the function consistent. As for dependencies, we can put `[setIsValid, setTempProjectImage]` there. The file comes from the event, so there's no real external data.

Reply:
Ok got it thank you :)